src/sources/azure_blob/queue.rs

```diff
@@ -160,57 +161,69 @@ async fn proccess_event_grid_message(
         );
         return None;
     }
-    let parts = body.subject.split("/").collect::<Vec<_>>();
-    if parts.len() != 7 {
-        warn!("Ignoring event because of wrong subject format");
-        return None;
-    }
-    let container = parts[4];
-    if container != container_client.container_name() {
-        // This shouldn't happen if everything is configured properly.
-        return None;
-    }
-    let blob = parts[6];
-    info!(
-        "New blob created in container '{}': '{}'",
-        &container, &blob
-    );
+    match parse_subject(body.subject) {
+        Some((container, blob)) => {
+            if container != container_client.container_name() {
+                trace!("Container name does not match. Skipping.");
```

Comment:
This should be either an error or a warning. It's not a sunny day path.

Reply:
Well, in our setup it is acceptable. Event subscriptions are per storage account, not container. So we can get events for other containers as well. IMO having it as trace is ok.